packages/lib-react-components/src/Markdownz/Markdownz.spec.js

```diff
@@ -11,4 +13,257 @@ describe('<Markdownz />', function () {
+      expect(url).to.equal('')
+    })
+
+    it('should return an empty string if the resource is falsy, but the mention symbol is', function () {
+      const url = wrapper.instance().buildResourceURL('', '@')
+      expect(url).to.equal('')
+    })
+
+    describe('when the symbol is #', function () {
+      it('should return the expected url with a search query param', function () {
+        const resource = 'tigers'
+        const baseURL = ''
+        const expectedReturnValue = `${baseURL}/talk/search?query=${resource}`
+        wrapper.instance().buildResourceURL(resource, '#')
+        expect(buildResourceURLSpy).to.have.been.calledOnce
+        expect(buildResourceURLSpy).to.have.returned(expectedReturnValue)
+      })
+
+      it('should return a url with the project slug if props.projectSlug is defined', function () {
+        const resource = 'tigers'
+        const projectSlug = 'zooniverse/snapshot-wakanda'
+        const expectedReturnValue = `/projects/${projectSlug}/talk/tag/${resource}`
+        wrapper.setProps({ projectSlug })
+        wrapper.instance().buildResourceURL('tigers', '#')
+        expect(buildResourceURLSpy).to.have.been.calledOnce
+        expect(buildResourceURLSpy).to.have.returned(expectedReturnValue)
+        wrapper.setProps({ projectSlug: '' })
+      })
+
+      it('should return a url with props.baseURI if defined', function () {
+        const resource = 'tigers'
+        const baseURI = 'https://classrooms.zooniverse.org'
+        const expectedReturnValue = `${baseURI}/talk/search?query=${resource}`
+        wrapper.setProps({ baseURI })
+        wrapper.instance().buildResourceURL('tigers', '#')
+        expect(buildResourceURLSpy).to.have.been.calledOnce
+        expect(buildResourceURLSpy).to.have.returned(expectedReturnValue)
+        wrapper.setProps({ baseURI: '' })
+      })
+    })
+
+    describe('when the symbol is @', function () {
+      it('should return a url for the user profile', function () {
+        const resource = 'srallen'
+        const expectedReturnValue = `/users/${resource}`
+        wrapper.instance().buildResourceURL(resource, '@')
+        expect(buildResourceURLSpy).to.have.been.calledOnce
+        expect(buildResourceURLSpy).to.have.returned(expectedReturnValue)
+      })
+
+      it('should return a url with the project slug if props.projectSlug is defined', function () {
+        const resource = 'srallen'
+        const projectSlug = 'zooniverse/snapshot-wakanda'
+        const expectedReturnValue = `/projects/${projectSlug}/users/${resource}`
+        wrapper.setProps({ projectSlug })
+        wrapper.instance().buildResourceURL(resource, '@')
+        expect(buildResourceURLSpy).to.have.been.calledOnce
+        expect(buildResourceURLSpy).to.have.returned(expectedReturnValue)
+        wrapper.setProps({ projectSlug: '' })
+      })
+
+      it('should return a url with props.baseURI if defined', function () {
```

Comment:
You're repeating the same test functions for each symbol, so consider abstracting the contents of these `describe` blocks into a function that checks the three expectations (query param, project slug and base URI) for a given symbol and resource.

Reply:
There's a slight variation in a couple of the tests, but I can do this if you feel it's important.

Reply:
It's no big deal. I thought it would save on some maintenance if test functions were being repeated.